shared/actions/kbfs/index.platform.desktop.js

```diff
@@ -48,55 +49,70 @@ function formKbfsPathWindows (extendedConfig: ExtendedStatus): string {
+    kbfsPath = null
+
+    // there can be a race between extendedConfig / kbfs / everything loading so we try a couple of times
+    for (let i = 0; i < 3; ++i) {
+      const extendedConfig = yield select(state => state.config.extendedConfig)
+      try {
+        kbfsPath = yield call(windowsKBFSRoot, extendedConfig)
+        if (kbfsPath) {
+          break
+        }
+      } catch (error) {
+        kbfsPathError = error
+      }
+
+      yield put(getExtendedStatus())
+      yield race({
+        loaded: take(Constants.extendedConfigLoaded),
+        timeout: call(delay, 1000),
+      })
+    }
+
+    if (!kbfsPath) {
+      console.warn('Error in parsing kbfsPath: ', kbfsPathError)
       return
     }
-    openPath = openPath.slice(Constants.defaultKBFSPath.length)
-
-    const state = getState()
-    const kbfsPath = state.config.kbfsPath
-
-    // On windows the path isn't /keybase
-    // We can figure it out by looking at the extendedConfig though
-    if (kbfsPath === Constants.defaultKBFSPath) {
-      const extendedConfig = state.config.extendedConfig
-      const extendedConfigPromise = extendedConfig
-        ? Promise.resolve(extendedConfig)
-        : Promise.reject()
-
-      extendedConfigPromise
-        .then(formKbfsPathWindows)
-        // In case the first try fails, let's try to get the extendedConfig again
-        // This can happen because kbfs loads after the first getExtended status call
-        // $FlowIssue
-        .catch(() => dispatch(getExtendedStatus()).then(formKbfsPathWindows))
-        .then(kbfsPath => {
-          dispatch({type: Constants.changeKBFSPath, payload: {path: kbfsPath}})
-
-          openPath = path.resolve(kbfsPath, openPath)
-          // Check to make sure our resolved path starts with the kbfsPath
-          // i.e. (not opening a folder outside kbfs)
-          if (!openPath.startsWith(kbfsPath)) {
-            throw new Error(`openInKBFS requires ${kbfsPath} prefix: ${openPath}`)
-          }
-          open(openPath)
-        })
-        .catch(e => {
-          console.warn('Error in parsing kbfsPath:', e)
-        })
-    } else {
-      open(path.resolve(kbfsPath, openPath))
+
+    yield put({type: Constants.changeKBFSPath, payload: {path: kbfsPath}})
+  }
+
+  if (kbfsPath) {
+    openPath = path.resolve(kbfsPath, openPath)
+    // Check to make sure our resolved path starts with the kbfsPath
+    // i.e. (not opening a folder outside kbfs)
+    if (!openPath.startsWith(kbfsPath)) {
+      throw new Error(`openInKBFS requires ${kbfsPath} prefix: ${openPath}`)
     }
-  })
+    open(openPath)
+  }
 }
 
-export function openInKBFS (openPath: string = Constants.defaultKBFSPath): AsyncAction {
+function * openSaga (action: FSOpen): SagaGenerator<any, any> {
+  const openPath = action.payload.path || Constants.defaultKBFSPath
+
   console.log('openInKBFS:', openPath)
   if (process.platform === 'win32') {
-    return openInWindows(openPath)
+    yield * openInWindows(openPath)
```

Comment:
http://yelouafi.github.io/redux-saga/docs/api/index.html#callfn-args

When you do `call` you aren't actually running the function right away, you are telling the middleware to do it, and it behaves in the way documented above. So it lets the called generator finish before resuming.

I found these docs that explain this better:
http://yelouafi.github.io/redux-saga/docs/advanced/SequencingSagas.html
http://yelouafi.github.io/redux-saga/docs/advanced/ComposingSagas.html

tl;dr yield* works but it's more composable to do `call`